src/connector/builder.rs

```diff
@@ -44,6 +46,27 @@ impl ConnectorBuilder<WantsTlsConfig> {
     /// [`enable_http2`](ConnectorBuilder::enable_http2)) before the
     /// connector is built.
     pub fn with_tls_config(self, config: ClientConfig) -> ConnectorBuilder<WantsSchemes> {
+        assert!(
```

Comment:
I'd propose we instead take `impl Into<Arc<ClientConfig>>` for `with_tls_config()`, which can cover both cases?

Reply:
That would probably be ideal; I was trying to avoid making a breaking change but I could do that if it was preferred.

Reply:
Ahhhh I see what you are saying. I did not realize that the stdlib implemented [`From<T> for Arc<T>`](https://doc.rust-lang.org/std/sync/struct.Arc.html#impl-From%3CT%3E-for-Arc%3CT%3E).